Add missing STD Library Identifiers

wienans opened the pull request:
Added a STDLIB_IDENTIFIERS for C11 as well as updated the C99 once based on,

C99: https://www.dii.uchile.cl/~daespino/files/Iso_C_1999_definition.pdf
C11: https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1570.pdf

in the misra.py

danmar reviewed (commented):
hmm.. some tests would be nice. Do you have some idea? I don't feel we need to test this in full but some kind of generic tests would be nice.. maybe ensure `C11_STDLIB_IDENTIFIERS` extends `C99_STDLIB_IDENTIFIERS`, make a few `isStdLibId` calls to make sure it works, ..

wienans commented:
> hmm.. some tests would be nice. Do you have some idea? I don't feel we need to test this in full but some kind of generic tests would be nice.. maybe ensure `C11_STDLIB_IDENTIFIERS` extends `C99_STDLIB_IDENTIFIERS`, make a few `isStdLibId` calls to make sure it works, ..

@danmar Test suggestions sound good, I can add that. is there already a test file where the misra.py is „unit tested“? I only found the .c file which tests the detections. If yes could you link the file where I should add it, if no could you make a suggestion how you would like to have the test added?

with test suggestions I don’t have additional ideas, the thing is it’s only a big list of keywords so there is not much to test, apart of the suggestions you already mentioned.

wienans commented:
@danmar added a Test and also checked the isKeyword additionally to isStdLib and fixed a typo there, too.
Additionally i changed the switching std identifieres for C11 from a specific C11 check to else. Therefore all standards after C11 will get the C11 Std Lib Identifiers and Keywords.

If it is fine from your side it can be merged.

danmar commented:
@wienans thanks! I wonder, is your name added in https://github.com/danmar/cppcheck/blob/main/AUTHORS yet?

wienans commented:
@danmar actually no i am not in it. You can add me as Sven Wienand
